gateway/gateway.go

```diff
@@ -29,14 +30,17 @@ type Gateway struct {
 	Message        chan config.Message
 	Name           string
 	Messages       *lru.Cache
+	MessageStore   gokv.Store
+	CanonicalStore gokv.Store
 
 	logger *logrus.Entry
 }
 
 type BrMsgID struct {
-	br        *bridge.Bridge
-	ID        string
+	Protocol  string
```

Comment:
We are directly storing these values in our persistent store, storing a reference to br, would just be garbage data when we read it back.
Based on the current usage of `br` for this struct we are only using it to get the protocol and destination gateway name. 

https://github.com/42wim/matterbridge/blob/8587fa85854b8ba14a171edf1c9c7e8f066eb6fb/gateway/gateway.go#L267-L269

https://github.com/yousefmansy1/matterbridge/blob/c0f5d0c5f7a5c7c9edd871bea1cc07ebadbcdb1a/gateway/gateway.go#L313-L315

For this persistent feature we are removing this reference and storing the direct values.
Unfortunately, that does mean if you change the name of the gateway things will break.

Reply:
Sure, I understand not storing the `br` reference it self, but you can still reference `br.Protocol` and `br.Name` instead of creating those 2 variables? Or am I missing something about this?

Reply:
For the in memory cache yes.
But for the persistent message map, if we restart the application all the memory references we have stored inside our value store will point whatever is the old `br` reference, which would be different for each run of the application.

I could be misunderstanding something about how references work in golang, correct me if I'm wrong.

Reply:
if you actually store `br` it's a reference, but `br.Protocol` or `br.Name` is a string

Reply:
Oh ok I see. Yes, that would make more sense and would make its functionality more clear.
Will update.

Reply:
Actually maybe I'm misunderstanding

```go
type BrMsgID struct {
	Protocol  bridge.Bridge.Protocol
	DestName  bridge.Bridge.Name
	ChannelID string
	ID        string
}
```

causes build errors:
```
yousef@DESKTOP-YOUSEF $ go build -tags whatsappmulti -gcflags=all="-N -l"
# github.com/42wim/matterbridge/gateway
gateway/gateway.go:40:25: syntax error: unexpected . in struct type; possibly missing semicolon or newline or }
```

Reply:
BrMsgID stays the same with br *bridge.Bridge, where you need the protocol just use br.Protocol
I'm doing this on mobile so I could also be misunderstanding your issue

Reply:
I might also be misunderstanding what you mean, but this change is most important in this code snippet

https://github.com/yousefmansy1/matterbridge/blob/60219a39d1624df64de9421177b326bde269319f/gateway/persistent.go#L59-L74

When we get the `[]BrMsgID` from the message store, anything that we read from those items cant come from any reference.